src/main/java/org/opensearch/knn/plugin/stats/KNNFlag.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.plugin.stats;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Class defines collection of flags for stats
+ */
+public enum KNNFlag {
+    BUILT_WITH_FAISS,
+    BUILT_WITH_LUCENE,
+    BUILT_WITH_NMSLIB;
+
+    private final AtomicBoolean flag = new AtomicBoolean(false);
```

Comment:
why this atomic?

Reply:
We probably don't need this, initially I was thinking of different implementation we more detailed stats, this is the artifact that left from that idea. With boolean flag concurrent access doesn't matter match, let me change it to a primitive boolean